libs/common/RespLengthEncodingUtils.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Linq;
+
+namespace Garnet.common;
+
+/// <summary>
+/// Utils for working with RESP length encoding
+/// </summary>
+public static class RespLengthEncodingUtils
+{
+    /// <summary>
+    /// Decodes the RESP-encoded length and returns payload start
+    /// </summary>
+    /// <param name="buff"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentException"></exception>
+    public static (long length, byte payloadStart) DecodeLength(ref ReadOnlySpan<byte> buff)
+    {
+        // remove the value type byte
+        var encoded = buff.Slice(1);
+
+        if (encoded.Length == 0)
+        {
+            throw new ArgumentException("Encoded length cannot be empty.", nameof(encoded));
+        }
+
+        var firstByte = encoded[0];
+        return (firstByte >> 6) switch
+        {
+            // 6-bit encoding
+            0 => (firstByte & 0x3F, 1),
+            // 14-bit encoding
+            1 when encoded.Length < 2 => throw new ArgumentException("Not enough bytes for 14-bit encoding."),
+            1 => (((firstByte & 0x3F) << 8) | encoded[1], 2),
+            // 32-bit encoding
+            2 when encoded.Length < 5 => throw new ArgumentException("Not enough bytes for 32-bit encoding."),
+            2 => ((long)((encoded[1] << 24) | (encoded[2] << 16) | (encoded[3] << 8) | encoded[4]), 5),
+            _ => throw new ArgumentException("Invalid encoding type.", nameof(encoded))
+        };
+    }
+
+    /// <summary>
+    /// Encoded payload length to RESP-encoded payload length
+    /// </summary>
+    /// <param name="length"></param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentOutOfRangeException"></exception>
+    public static byte[] EncodeLength(long length)
```

Comment:
This whole method is pretty allocate-y, which worries me if it's ever extensively used.

Reply:
I agree bit @kevin-montrose here. A common approach is to have small static method to first calculate the required buffer size (it can be simple switch expression) for the caller (i.e. in our case ``NetworkDUMP`` method) to know how much buffer it needs (to slice from existing buffer, stackalloc, rent etc., doesn't matter).

Then the actual encoding method takes a ``Span<byte> output`` argument which ownership belongs to caller. Now we could do the encoding on this buffer without allocations. You can see this pattern all over .NET with ``TryWrite`` methods.